python/astro_metadata_translator/translators/helpers.py

```diff
@@ -129,3 +132,61 @@ def tracking_from_degree_headers(self, radecsys, radecpairs, unit=u.deg):
     if self.to_observation_type() == "science":
         raise KeyError("Unable to determine tracking RA/Dec of science observation")
     return None
+
+
+def altaz_from_degree_headers(self, altazpairs, obstime, is_zd=None):
+    """Calculate the altitude/azimuth coordinates from lists of headers.
+
+    If the altitude is found but is greater than 90 deg, it will be returned
+    fixed at 90 deg.
+    If the altitude or azimuth are negative and this is a calibration
+    observation, `None` will be returned.
+
+    Parameters
+    ----------
+    altazpairs : `tuple` of `str`
+        Pairs of keywords specifying Alt/Az in degrees. Each pair is tried
+        in turn.
+    obstime : `astropy.time.Time`
+        Reference time to use for these coordinates.
+    is_zd : `set`, optional
+        Contains keywords that correspond to zenith distances rather than
+        altitude.
+
+    Returns
+    -------
+    altaz = `astropy.coordinates.AltAz`
+        The AltAz coordinates associated with the telescope location
+        and provided time.  Returns `None` if this observation is not
+        a science observation and no AltAz keys were located.
+
+    Raises
+    ------
+    KeyError
+        No AltAz keywords were found and this observation is a science
+        observation.
+    """
+    for alt_key, az_key in altazpairs:
+        if az_key in self._header and alt_key in self._header:
+            az = self._header[az_key]
+            alt = self._header[alt_key]
+
+            # Check for zenith distance
+            if is_zd and alt_key in is_zd:
+                alt = altitude_from_zenith_distance(alt * u.deg).value
+
+            if az < 0.0 or alt < 0.0:
```

Comment:
Can `az` not be negative.  I understand `alt` not going below the horizon, but `az` has a natural periodic boundary condition.

Reply:
Hmm. I guess that depends on the telescope. Would -180 be a reasonable value?

Reply:
I think CFHT triggered this because they use -9999 or something to indicate no value.

Reply:
That's unfortunate.  Either -180 or -360.  Your choice.